chain-core/src/init/config.rs

```diff
@@ -309,44 +293,29 @@ impl InitConfig {
         }
     }
 
-    fn is_rewards_pool_address(&self, address: &RedeemAddress) -> bool {
-        *address == self.launch_incentive_from
-            || *address == self.launch_incentive_to
-            || *address == self.long_term_incentive
-    }
-
-    /// returns the initial accounts and rewards pool state
+    /// returns the initial accounts
     /// assumes one called [validate_config_get_genesis], otherwise it may panic
-    fn get_genesis_state(
-        &self,
-        genesis_time: Timespec,
-        validator_addresses: HashSet<RedeemAddress>,
-    ) -> (Vec<StakedState>, RewardsPoolState) {
-        let mut rewards_pool_amount: u64 = 0;
-        let mut accounts = Vec::with_capacity(self.distribution.len());
-        for (address, (amount, address_type)) in self.distribution.iter() {
-            if self.is_rewards_pool_address(address) || *address_type == AccountType::Contract {
-                rewards_pool_amount += u64::from(*amount);
-            } else {
-                accounts.push(StakedState::new_init(
+    fn get_account(&self, genesis_time: Timespec) -> Vec<StakedState> {
+        self.distribution
+            .iter()
+            .map(|(address, (destination, amount))| {
+                let bonded = match destination {
+                    StakedStateDestination::Bonded => true,
+                    StakedStateDestination::UnbondedFromGenesis => false,
+                    StakedStateDestination::UnbondedFromCustomTime(_time) => false,
```

Comment:
`UnbondedFromCustomTime` is ignored -- so I guess it'll be implemented in a different PR with `new_init` changes?

Reply:
yes, I think so.